retry.go

```diff
@@ -0,0 +1,105 @@
+package aviation
+
+import (
+	"context"
+	"time"
+
+	"github.com/jpillora/backoff"
+	"github.com/mongodb/grip"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+)
+
+func MakeRetryUnaryClientInterceptor(maxRetries int) grpc.UnaryClientInterceptor {
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		var lastErr error
+		b := &backoff.Backoff{
+			// TODO: figure out the best options for this.
+			Min:    100 * time.Millisecond,
+			Max:    time.Second,
+			Factor: 2,
+		}
+		callCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
```

Comment:
is this really needed?

Reply:
We concluded a few days ago to have a separate calling context in order not to litter the parent context. I can get rid of it if it is not necessary?

Reply:
I think it's not necessary